telegram/base.py

```diff
@@ -80,3 +80,25 @@ def to_dict(self):
                     data[key] = value
 
         return data
+
+    def _get_id(self):
+        for x in ['id', 'file_id', 'message_id', 'result_id', 'update_id']:
+            if hasattr(self, x):
+                return self.__getattribute__(x)
+        raise NotImplementedError
+
+    def __eq__(self, other):
+        if isinstance(other, self.__class__):
+            try:
+                return self._get_id() == other._get_id()
```

Comment:
I'd say it's okay since I'm checking the type of the class first, so I know with almost 100% certainty how `_get_id()` operates.
I seem to recall having read somewhere that basing `__eq__` on hash is very discouraged.